src/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/UnprotectedAttributeTests.cs

```diff
@@ -268,6 +270,43 @@ private static void VerifyUnprotectedAttributes_Heterogenous3(byte[] encodedMess
         }
 
         [Fact]
+        public static void TestUnprotectedAttributes1_EmptySet()
+        {
+            // This tests the behavior of unprotected attribute extraction when one of the attribute sequences declares an
+            // attribute type, but the contained SET OF AttributeValue is empty.
+            // Attribute ::= SEQUENCE {
+            //      attrType OBJECT IDENTIFIER,
+            //      attrValues SET OF AttributeValue }
+            // The encoded message was built in ASN.1 editor and tested in framework.It contains an enveloped message
+            // version 2 with a key transport recipient, the enveloped message contains data encrypted with 3DES.
+            // The attributes set is built as
+            // {
+            //      { attrType: document description, attrValues: {} },
+            //      { attrType: document name, attrValue: { } },
+            // }
+            // The important part of this test is that there are 0 attributes of a type that is declared within the encoded message.
+            // On framework, this would return 2 as explained on issue 9252, that is it woulds create a CryptographicObjectCollection
+            // with two CryptographicObject, the first one holding a list of document description with the two values, the second one
+            // holding an empty list of document name. This makes ecms.UnprottectedAttributes.Count be 2. Core doesn't create the second
+            // list and as such ecms.UnprottectedAttributes.Count returns 1 as documented here.
```

Comment:
Each CryptographicAttributeObject works as a data structure that holds an AsnEncodedDataCollection for a given OID, and as there are two objects with different OIDs there should be two CryptographicAttributeObject elements in the collection. One that holds an empty collection and one that holds two. ecms.UnprotectedAttributes.Count only reports the number of CryptographicAttributeObject, not the amount of AsnEncodedData that the overall structure holds.